### PR TITLE
JAMES-3680 oidc docker-compose: SSO

### PR DESCRIPTION
**Pull request by Arsnael**
Still a WIP. 

Got so far a preprovisionned Keycloak with a ldap and a realm, and james memory as well where you need to run a provisioning script to match with the user on the ldap.

### CODE REVIEW

**Comment by chibenwa** [comment]
Don't forget README, 

IMO it should include sample curl commands to:

 - generate an access token
 - List JMAP mailboxes using that access token
 - calling the logout endpoint.

File: examples/oidc/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+version: '3'
+
+services:
+
+  james:
+    image: apache/james:memory-3.6.1
```
Comment (chibenwa) [inline examples/oidc/docker-compose.yml:6]: Won't support the X-User stuff. Please use `latest` or `3.7.0` to make it clear that the latest master is required for this to work.

Comment (Arsnael) [reply examples/oidc/docker-compose.yml:6]: Good remark

**Comment by chibenwa** [comment]
We still need Krakend container and SLO side container, correct?

**Comment by Arsnael** [comment]
> We still need Krakend container and SLO side container, correct?

Correct

**Comment by Arsnael** [comment]
I have an issue with krakend and keycloak locally... when I try to use an oidc token against the endpoint I get a weird error:
```
krakend     | [GIN] 2022/01/11 - 09:49:48 | 401 |    3.194549ms |      172.19.0.1 | POST     "/oidc/jmap"
krakend     | Error #01: Get "keycloak/auth/realms/oidc/protocol/openid-connect/certs": unsupported protocol scheme ""
```
Will dig tomorrow... I'm not sure if it's linked to the local conf or something in the realm

**Comment by Arsnael** [comment]
SSO part ready to review

**Review by vttranlina — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

File: examples/oidc/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+However, you can use the Keycloak playground example:
+
+* Open your browser and go to https://www.keycloak.org/app/
+* Fill the form with your local Keycloak info and click `Save`:
+  * Keycloak URL: `http://localhost:8080/auth`
+  * Realm: `oidc`
+  * Client: `oidc`
+* Click `Sign in` and you will get redirected to your Keycloak login screen
+* Open the Developer Tools of your browser and check the network connections
+* Enter the credentials of the user: `james-user@localhost / secret`
```
[the remark this answers is not in the document]
Comment (Arsnael) [reply examples/oidc/README.md:36]: Oh damn I didn't know that one on Postman. Thanks for the tip, will come very handy :D

[the remark this answers is not in the document]
Comment (chibenwa) [reply examples/oidc/README.md:36]: Agreed, we could use postman, it likely is easier :-) (or we could put both, or we could have curl + postman screenshot, etc...)

[the remark this answers is not in the document]
Comment (Arsnael) [reply examples/oidc/README.md:36]: I'm not sure though... Can we revoke the token from there or not? Or it will be a problem for SLO :)

**Pull request merged**